.github/workflows/build.yml

```diff
@@ -0,0 +1,77 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  schedule:
+    - cron: '0 5 * * *'
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      max-parallel: 3
+      fail-fast: false
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+        node: [12]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+
+      - name: Use Node.js ${{ matrix.node }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: Build
+        shell: bash
+        run: |
+          yarn install --no-lockfile
+          yarn prepare
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NODE_OPTIONS: "--max_old_space_size=4096"
+
+      - name: Test
+        shell: bash
+        run: npx lerna run --concurrency=1 --stream --scope "@theia/!(example-)*" test
```

Comment:
Why not use the script:

https://github.com/eclipse-theia/theia-cpp-extensions/blob/bb6bb9dfbc36daee0315f6f511c4c4b0fb801eef/package.json#L43-L44

- [upstream changes](https://github.com/eclipse-theia/theia/pull/8780/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R65)

Reply:
We cannot use `yarn test:theia` yet until the fix is submitted upstream.
Currently, this approach causes the `macOS` build to fail.